### PR TITLE
K8SPXC-1407 Clarify HA Proxy upgrade risks

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: docs/update.md [hunk 1]
```diff
@@ -182,9 +182,12 @@ Manual update of Percona XtraDB Cluster can be done as follows:
 
     !!! warning
 
-        The above command upgrades various components of the cluster including PMM Client. It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server **before** upgrading PMM Client. If it wasn't done and you would like to avoid PMM Client upgrade, remove it from the list of images, reducing the last of two patch commands as follows:
-        
-        
+        The above command upgrades various components of the cluster including PMM Client and HA Proxy, which may need additional attention.
+
+        * If you are using [custom configuration for HA Proxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), check [HA Proxy release information](https://www.haproxy.org/) to make sure that the version you are applying is compatible with your existing custom config.
```
Comment (hors) [inline docs/update.md:187]: @fiowro, the issue is that we occasionally add new options to the HAProxy configuration. If users are using a custom configuration for HAProxy, they need to check the default configuration at https://github.com/percona/percona-docker/blob/pxc-operator-1.14.0/haproxy/dockerdir/etc/haproxy/haproxy-global.cfg and incorporate any new options into their custom config.

For example, in this commit: https://github.com/percona/percona-docker/commit/f2b9f63e8c288d219522ea76ae6b113e2717a59c, we added two new options in different places. If these options are not included in the custom config, users may experience downtime.

The instructions are as follows:
1. check new conf before the update
2. add new options while the update into CR

**Review by hors — CHANGES_REQUESTED** [review]

File: docs/update.md [hunk 2]
```diff
@@ -182,9 +194,8 @@ Manual update of Percona XtraDB Cluster can be done as follows:
 
     !!! warning
 
-        The above command upgrades various components of the cluster including PMM Client. It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server **before** upgrading PMM Client. If it wasn't done and you would like to avoid PMM Client upgrade, remove it from the list of images, reducing the last of two patch commands as follows:
-        
-        
+        The above command upgrades various components of the cluster including PMM Client. If you didn't follow the [official recommendation](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server before upgrading PMM Client, you can avoid PMM Client upgrade by removeing it from the list of images as follows:
```
Comment (egegunes) [inline docs/update.md:197]:
```suggestion
        The above command upgrades various components of the cluster including PMM Client. If you didn't follow the [official recommendation](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) to upgrade PMM Server before upgrading PMM Client, you can avoid PMM Client upgrade by removing it from the list of images as follows:
```

File: docs/update.md [hunk 3]
```diff
@@ -112,11 +112,23 @@ are updated. Smart Update strategy is on when the `updateStrategy` key in the
...
+!!! warning
+
+    The upgrade covers various components of the cluster including PMM Client
+    and HA Proxy, which may need additional attention.
+    
+    *  It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html)
+        to upgrade PMM Server **before** upgrading PMM Client.
+    
+    * If you are using [custom configuration for HA Proxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), 
+        **before upgrading**, check the HA Proxy configuration file provided by
```
Comment (egegunes) [inline docs/update.md:127]:
```suggestion
        **before upgrading**, check the HAProxy configuration file provided by
```

File: docs/update.md [hunk 4]
```diff
@@ -112,11 +112,23 @@ are updated. Smart Update strategy is on when the `updateStrategy` key in the
...
-    [appropriate section](update.md#more-on-upgrade-strategies).
+!!! warning
+
+    The upgrade covers various components of the cluster including PMM Client
+    and HA Proxy, which may need additional attention.
+    
+    *  It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html)
+        to upgrade PMM Server **before** upgrading PMM Client.
+    
+    * If you are using [custom configuration for HA Proxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy),
```
Comment (egegunes) [inline docs/update.md:126]:
```suggestion
    * If you are using [custom configuration for HAProxy](haproxy-conf.md#passing-custom-configuration-options-to-haproxy), 
```

File: docs/haproxy-conf.md [hunk 5]
```diff
@@ -83,8 +83,13 @@ You can pass custom configuration to HAProxy in one of the following ways:
 !!! note
 
     If you specify a custom HAProxy configuration in this way, the
-    Operator doesn’t provide its own HAProxy configuration file except [several hardcoded options](https://github.com/percona/percona-docker/blob/main/haproxy/dockerdir/etc/haproxy/haproxy.cfg) (which therefore can't be overwritten). That’s why you
+    Operator doesn’t provide its own HAProxy configuration file except [several hardcoded options](https://github.com/percona/percona-docker/blob/pxc-operator-{{ release }}/haproxy/dockerdir/etc/haproxy/haproxy.cfg) (which therefore can't be overwritten). That’s why you
     should specify either a full set of configuration options or nothing.
+    Additionally, when [upgrading Percona XtraDB Cluster](update.md#upgrading-percona-xtradb-cluster)
+    it would be wise to check the
+    [HA Proxy configuration file](https://github.com/percona/percona-docker/blob/pxc-operator-{{ release }}/haproxy/dockerdir/etc/haproxy/haproxy-global.cfg)
```
Comment (egegunes) [inline docs/haproxy-conf.md:90]:
```suggestion
    [HAProxy configuration file](https://github.com/percona/percona-docker/blob/pxc-operator-{{ release }}/haproxy/dockerdir/etc/haproxy/haproxy-global.cfg)
```

File: docs/update.md [hunk 6]
```diff
@@ -112,11 +112,23 @@ are updated. Smart Update strategy is on when the `updateStrategy` key in the
...
 
-    As an alternative, the `updateStrategy` key can be set to `RollingUpdate` 
-    and `OnDelete`. You can find out more about it in the
-    [appropriate section](update.md#more-on-upgrade-strategies).
+!!! warning
+
+    The upgrade covers various components of the cluster including PMM Client
+    and HA Proxy, which may need additional attention.
+    
+    *  It is [highly recommended](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html)
```
Comment (egegunes) [inline docs/update.md:123]:
```suggestion
    *  It is **highly recommended**
```

File: docs/update.md [hunk 7]
```diff
@@ -112,11 +112,23 @@ are updated. Smart Update strategy is on when the `updateStrategy` key in the
...
+        to upgrade PMM Server **before** upgrading PMM Client.
```
Comment (egegunes) [inline docs/update.md:124]:
```suggestion
        to [upgrade PMM Server](https://docs.percona.com/percona-monitoring-and-management/how-to/upgrade.html) **before** upgrading PMM Client.
```

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**